### ISSUE TITLE
Crash on FBX Import (Booth SIGNALIS Elster)

### ISSUE THREAD

**Issue by Enverex** [Yellow-Dog-Man/Resonite-Issues#168]
### Describe the bug?

Trying to import elster.fbx from this package results in Resonite hard crashing. It locks up for a few second then completely dies (no Unity crash handler).

### To Reproduce

Download the package from here - https://booth.pm/en/items/4913158 (free)
Then try and import elster.fbx from the extracted contents.

The settings I chose when importing were:
* Auto Humanoid Height
* Prefer Specular
* Import External Textures
* Import Textures by Name

### Expected behavior

For Resonite not to crash ;)

### Resonite Version Number

2023.10.13.743

### What Platforms does this occur on?

Windows

### Log Files

[Player.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/12907010/Player.log)

Player log doesn't have anything useful and no Unity Crashlog appears to be generated so no useful logs. But the FBX is provided so hopefully that's enough on it's own to track down the issue.

**Comment by troyBORG** [comment]
Do you know did the avatar creator use Maya by chance?
I know the creator of the [Zorrix](https://wuffwick.gumroad.com/l/Zorrix) I use.  When you import their FBX it did the same thing.

Because they used a "newer" version of the FBX format 

I know on the old platform wiki it said:
Autodesk ( .fbx ) - Must be FBX 2011, 2012 or 2013, newer ones won't work.

I had to convert the model from [FBX to GLTF](https://github.com/facebookincubator/FBX2glTF) then import it would import into blender without the bones going phsyco,  so I can edit it and then export it as FBX and that worked.

**Comment by Enverex** [comment]
I have no idea what was used to create it unfortunately. The game hard-crashing is obviously an issue regardless.

**Comment by kazu0617** [comment]
@troyBORG 

I certainly think that's what I'll do when I do my research, and I'll do that when I do my research.
But more than that, I think the condition of "crashing reliably when an unsupported FBX comes in" is a big problem, especially when setting up avatars for beginners. In fact, if it crashes, it increases the likelihood that they will quit the game.
So when an unsupported FBX comes in, I'd like you to stop the importer with a proper "unsupported FBX" and I hope they stop the importer.

---

確かに調査するとそうなると思うし、自分が調べるときはそうするよ。
ただ、それ以上に「サポート外のFBXが入ったときに、確実にCrushする」状態は特に初心者にアバターをセットアップしてもらうときには大きな問題になると思う。実際クラッシュしてしまうとそこでゲームをやめてしまう可能性が高まるからね。
なので、サポート外のFBXが来たときにきちんと「対応していないFBXだよ！」とインポータを止めて欲しいな。

**Comment by GearBell** [comment]
I keep crashing as well with fbx imports. GTLF works just fine, though it does odd things for the blendshapes (not important to this). So far importing the fbx into a 3.0 blender then reexporting it works, however it rearranges the blendshapes, making a mesh transfer unusable. I am able to reporduce it consistantly, however it is with a private avatar, so I will only provide the raw file to a dev if they request it. So far it is consistant if I use any blender older than 3.0.

**Comment by Frooxius** [comment]
This looks like an Assimp bug, I put this file into Assimp's unit test and it fails: 
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/8838625/fb667526-d9cd-44ba-8a90-567a73fa155f)

We'll have to make a report to them with the model file. If you want, you can report it yourself, otherwise it might take a bit before we make the report ourselves.

**Comment by GearBell** [comment]
Is there a bandaid patch to roll back to the older assimp library until this is fixed (Im only quoting what I have heard from others with that piece)? I cannot import my own avatars or most other work until this is fixed.

**Comment by troyBORG** [comment]
> This looks like an Assimp bug, I put this file into Assimp's unit test and it fails: ![image](https://user-images.githubusercontent.com/8838625/283190996-fb667526-d9cd-44ba-8a90-567a73fa155f.png)

Where do I get that tester.  The FBX file that was for my Model (Which I think is just a newer FBX format) that crashes the game as well.
I wanna see what it reports about it.

**Comment by Frooxius** [comment]
@troyBORG It's part of the Assimp project: https://github.com/assimp/assimp

@GearBell Is this a regression? Do FBX files that worked before not work anymore? Are you able to provide me any samples for these?

**Comment by GearBell** [comment]
@Frooxius Yes the fbx files worked previously - I am currently wearing the latest version which is about a month old. It just crashes me now to import the fbx. I will send you the raw FBX in DMs - it is a nonpublic avatar though so please delete it when done testing.

**Comment by TisFoolish** [comment]
> I cannot import my own avatars or most other work until this is fixed.

Why not export them as gltf/glb and then import them as that. It's a significantly better and easier to support format

**Comment by troyBORG** [comment]
> @troyBORG It's part of the Assimp project: https://github.com/assimp/assimp

I tried and I don't get those error reporting messages mine only shows this:
![image](https://github.com/Yellow-Dog-Man/Resonite-Issues/assets/10406330/e66e340b-1387-4566-bb2d-b9cb365ecce9)

**Comment by Enverex** [comment]
Yeah, it fails on it for me:

``` assimp info elster.fbx
Launching asset import ...           OK
Validating postprocessing flags ...  OK
double free or corruption (!prev)
Aborted (core dumped)```

**Comment by GearBell** [comment]
> > I cannot import my own avatars or most other work until this is fixed.
> 
> Why not export them as gltf/glb and then import them as that. It's a significantly better and easier to support format

Both DAE and GTLF for some reason are no longer importing with blendshapes. They were working yesterday, though gtlf slightly corrupted the blendshapes (the body moves forward per blendshape, stacking makes the whole character move for some reason), but today blender files, dae, and gtlf are all importing with no blendshapes.

**Comment by GearBell** [comment]
So, this is new - I imported an fbx model that has no blendshapes, its a simple prop house off of sketchfab - it crashed me. There is no bones or blendshapes. Its just a downloaded model. Did not crash me when putting into blender and exporting as obj. I thought it is worth reporting that even prop fbx's now have a chance to crash you.

**Comment by GearBell** [comment]
Should I keep reporting imports that crash? Or is there enough to go off of that its on the list of bugs being looked at? I dont want to over-spam this.

**Comment by Frooxius** [comment]
@GearBell I mainly just need to collect files that cause the crash so I can test and potentially make a report. 

If you have any files that cause crash that you'd be okay sharing publicly, those would be most useful - we can send them to developers of Assimp.

**Comment by troyBORG** [comment]
I wonder if we can do the same thing for the [Zorrix Model](https://wuffwick.gumroad.com/l/Zorrix)?  
Its a paid avatar though.  So not sure not sure how we can get it to the Developers?

**Comment by Frooxius** [comment]
If anyone here is knowledgable with C++ and has access to models that are not shareable (or even if you want to do it for models you can) and willing to throw a bunch of time at this, you could try to make fix to Assimp and open PR on our fork: https://github.com/Yellow-Dog-Man/assimp

If anybody's willing to make the fix, I'll merge it quick and it can be contributed upstream too.

**Comment by troyBORG** [comment]
I figured out the problem with mine:

```
 .\assimp.exe info D:\Avatars\Zorrix\Models\ZorrixFBX.fbx
Launching asset import ...           OK
Validating postprocessing flags ...  OK
0 %
50 %
ERROR: Failed to load file: Validation failed: aiMesh::mBones[40], name = "WobbleJoint_Mane_Descending3_2" has the same name as aiMesh::mBones[55]
assimp info: Unable to load input file D:\Avatars\Zorrix\Models\ZorrixFBX.fbx
```

the mane wobble 4_2 was named 3_2 so there were 2 joints of the same name in different roots.   

Fixing this issue causes the model to import correctly.

**Comment by Krzeszny** [comment]
[DESKTOP-VHA2V7H - 2024.2.5.848 - 2024-02-06 14_20_27.log](https://github.com/Yellow-Dog-Man/Resonite-Issues/files/14185851/DESKTOP-VHA2V7H.-.2024.2.5.848.-.2024-02-06.14_20_27.log)
Here's what I get when I try to import 4 different FBX's (Fennix v4, Remaster, HD Planti, HD Digi), including one exported from Blender 4.0 (Fennix v7's FBX works though)
GLT has no problems.

**Comment by GearBell** [comment]
(I skimmed a lot so forgive me if it was already mentioned) I found the culprit to old fbx's causing crashing - its the blendshapes. If an old fbx doesnt have any itll import just fine, but if it does have blendshapes its best to import into an up to date 3d program then re-export as an fbx (or anything else you'd use) - either way I found the crashing factor. Not sure if this helps anything.
